compiler/noirc_evaluator/src/ssa/opt/loop_invariant.rs

```diff
@@ -93,11 +94,15 @@ struct LoopInvariantContext<'f> {
     // This map is expected to only ever contain a singular value.
     // However, we store it in a map in order to match the definition of
     // `outer_induction_variables` as both maps share checks for evaluating binary operations.
-    current_induction_variables: HashMap<ValueId, FieldElement>,
+    current_induction_variables: HashMap<ValueId, (FieldElement, FieldElement)>,
     // Maps outer loop induction variable -> fixed upper loop bound
     // This will be used by inner loops to determine whether they
     // have safe operations reliant upon an outer loop's maximum induction variable.
-    outer_induction_variables: HashMap<ValueId, FieldElement>,
+    outer_induction_variables: HashMap<ValueId, (FieldElement, FieldElement)>,
```

Comment:
The docs for both of these should be updated ("-> fixed lower and upper loop bound")